src/runtime/schema.ts

```diff
@@ -178,6 +182,24 @@ export class Schema {
     return Entity.createEntityClass(this, context);
   }
 
+  crdtConstructor<S extends Dictionary<Referenceable>, C extends Dictionary<Referenceable>>() {
+    const singletons = {};
+    const collections = {};
+    // TODO(shans) do this properly
+    for (const [field, {type}] of Object.entries(this.fields)) {
+      if (type === 'Text') {
+        singletons[field] = new CRDTSingleton<{id: string}>();
+      } else if (type === 'Number') {
+        singletons[field] = new CRDTSingleton<{id: string, value: number}>();
+      }
```

Comment:
Could you throw a big scary exception here, that makes it clear that the other types haven't been implemented yet?